### PR TITLE
DAOS-11032 build: Update SPDK to v22.01.2 maintenance release

### PR DESCRIPTION
**Pull request by tanabarr**
Update SPDK packaging to version 22.01.2 to pick up VMD-hotplug related changes.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Don't forget to update https://github.com/daos-stack/spdk/blob/a3a14f5ad6cc1744f09a4723b8bdffadd0b2e7d0/debian/control#L7

**Comment by tanabarr** [comment]
> Don't forget to update
> 
> https://github.com/daos-stack/spdk/blob/a3a14f5ad6cc1744f09a4723b8bdffadd0b2e7d0/debian/control#L7

Should we also update L34 from `libspdk20` to `libspdk22`?

**Comment by brianjmurrell** [comment]
> Should we also update L34 from `libspdk20` to `libspdk22`?

Hrm.  This really never should have been `libspdk20`.  It should have been `libspdk2`.  But indeed, now it should be `libspdk3` as the major version of `libspdk.so` is 3 now.  It was 2 before.

Hopefully our `libspdk20` package is long gone by the time they get their major version up to 20.

File: Jenkinsfile [hunk 1]
```diff
@@ -38,7 +38,7 @@
  */
 // To use a test branch (i.e. PR) until it lands to master
 // I.e. for testing library changes
-//@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@bmurrell/stop-testing-with-2.0") _
```
Comment (brianjmurrell) [inline Jenkinsfile:41]: This needs to be removed before it can land.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
dpdk PR needs to land first.

**Review by jolivier23 — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Comment by brianjmurrell** [comment]
> Hrm. This really never should have been `libspdk20`. It should have been `libspdk2`. But indeed, now it should be `libspdk3` as the major version of `libspdk.so` is 3 now. It was 2 before.
> 
> Hopefully our `libspdk20` package is long gone by the time they get their major version up to 20.

This still seems to be outstanding.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
libspdk20 -> libspdk3 for debian packaging still needs doing.

**Comment by tanabarr** [comment]
> > Hrm. This really never should have been `libspdk20`. It should have been `libspdk2`. But indeed, now it should be `libspdk3` as the major version of `libspdk.so` is 3 now. It was 2 before.
> > Hopefully our `libspdk20` package is long gone by the time they get their major version up to 20.
> 
> This still seems to be outstanding.

as described on Skype I tried this but there were multiple errors so I thought I would update that in a separate PR given its tangential to the purpose of this PR. does that sound okay @brianjmurrell . if you look at the previous builds you can see the failures

**Pull request merged**